l10n_it_account/models/l10n_it_account_mixin.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import models, fields
+
+
+class L10nItAccountMixin(models.AbstractModel):
+    _name = 'l10n_it_account.mixin'
+    _description = 'l10n_it_account Mixin'
+
+    is_company_it = fields.Boolean(
+        compute="_compute_is_company_it"
+    )
+
+    def _compute_is_company_it(self):
+        for record in self:
+            if self.env.user.company_id.is_country_id_code_it():
+                record.is_company_it = True
+            else:
+                record.is_company_it = False
```

Comment:
Simplify the expression, but also take into account the company of the object
```suggestion
            company = False
            if "company_id" in record._fields():
                company = record.company_id
            company = company or record.env.user.company_id
            record.is_company_it = company.is_country_id_code_it()
```